File: two_pointers/04_max_area/main.py
from typing import List

def bruteForce(heights:List[int]) -> int:
    res = 0
    for i in range(len(heights)):
        for j  in range(i + 1, len(heights)):
            res = max(res, min(heights[i], heights[j])) * (j - 1)
    return res 


def twoPointers(heights:List[int]) -> int:
    l, r = 0, len(heights) -1
    res = 0
    while l < r:
        area = min(heights[l], heights[r]) * (r - l)
        res = max(res, area)
        if heights[l] <= heights[r]:
            l += 1 
        else:
            r -= 1 
    return res

File: two_pointers/04_max_area/Main.java

public class Main {
    public int bruteForce(int[] heights){
        int res = 0;
        for(int i = 0; i<heights.length; i++){
            for (int j = i+1; j<heights.length; j++){
                res = Math.max(res, Math.min(heights[i], heights[j]) * (j - 1));
            }
        }
        return res;
    }


    public int twoPointers(int[] heights){
        int l = 0;
        int r = heights.length -1;
        int res = 0;

        while (l < r){
            int area = Math.min(heights[l], heights[r]) * (r - l);
            res = Math.max(res, area);
            if (heights[l] <= heights[r]){
                l++;
            }else{
                r--;
            }
        }
        return res;
    }
}
